### PR TITLE
Limit "all" option to Mac.

### PR DESCRIPTION
**Pull request by dlech**
Adding 100s of UUIDs to the scan filter on BlueZ causes BlueZ to fail
with:

    Traceback (most recent call last):
    File "examples/detection_callback.py", line 53, in <module>
        asyncio.run(main(service_uuids))
    File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
        return loop.run_until_complete(main)
    File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
        return future.result()
    File "examples/detection_callback.py", line 42, in main
        await scanner.start()
    File "/home/david/work/pybricks/bleak/bleak/backends/bluezdbus/scanner.py", line 169, in start
        assert_reply(reply)
    File "/home/david/work/pybricks/bleak/bleak/backends/bluezdbus/utils.py", line 23, in assert_reply
        raise BleakDBusError(reply.error_name, reply.body)
    bleak.exc.BleakDBusError: [org.bluez.Error.InProgress] Operation already in progress

`bluetoothd` has to be restarted to clear the error.

So don't allow this option on OS other than Mac.

Fixes: https://github.com/hbldh/bleak/issues/716

### CODE REVIEW

**Review by hbldh — APPROVED** [review]
Does the adding of this many uuids make scanning in Monterey slower than previous macOS versions? Regardless, it is needed for proper functionality...

Do we need to update and expand the list of uuids to cover devices not in that list? And in that case, where do we find them?

**Comment by dlech** [comment]
> Does the adding of this many uuids make scanning in Monterey slower than previous macOS versions?


It should not affect the "speed" at all. It is just a filter that excludes advertising data that doesn't match any of the UUIDs.

> Do we need to update and expand the list of uuids to cover devices not in that list? And in that case, where do we find them?

Getting the advertised UUIDs for every BLE device ever made seems like an impossible task. I would just leave it up to users to make a pull request to add a UUID if it is important to them.

**Comment by hbldh** [comment]
I agree. I dislike leaving difficult things like "knowing your device before connecting to it and seeing what's on it" to the user, but I don't think it is a viable option to try to include more known uuids in bleak without PRs.

Bleak on new macOS becomes less beginner-friendly than I would like. But this is as good as it gets, and hopefully good enough for most applications.

File: examples/detection_callback.py [hunk 1]
```diff
@@ -25,11 +25,18 @@ def simple_callback(device: BLEDevice, advertisement_data: AdvertisementData):
 
 
 async def main(service_uuids):
-    if len(service_uuids) > 0 and service_uuids[0] == "all":
-        # in Macos Monterey the service_uuids need to be specified.
+    if sys.platform == "darwin" and service_uuids and service_uuids[0] == "all":
```
Comment (mkoderer) [inline examples/detection_callback.py:28]: I would suggest that we enable this by default (without any parameter). And we use the logger.info block (maybe logging.warining would be better) below to give the user an hint that we only scan for predefined uuids.

Comment (dlech) [reply examples/detection_callback.py]: Updated.

**Comment by mkoderer** [comment]
Thanks David - LGTM

**Pull request merged**